docs/adr/0006-server-arch.md

```diff
@@ -0,0 +1,185 @@
+
+Below is a diagram and some example of how the new code architecture will work.
+It is imagined that for each domain (Submission, QC, Assign, Review, Decision)
+there will be a package that implements this pattern for that domain.
+
+<pre>
+
+    +------------------------------------------+
+    |            GraphQL Resolvers             |
+    +------------------------------------------+
+               |
+    +----------------------+       +-----------+
+    |      use cases       +<------+  config   |
+    +----------------------+       +-----------+
+        |           |                    |
+        |     +------------+       +-----------+
+        |     | aggregates |       | services  |
+        |     +------------+       +-----------+
+        |           |(n)
+    +------------------------------------------+
+    |                 models                   |
+    +------------------------------------------+
+    |             data access layer            |
+    +------------------------------------------+
+
+</pre>
+
+---
+
+## General Principles
+
+Each of the following sections describes the general principles that are to
+be used when writing code. This approach uses SOLID Principles.
+
+### GraphQL Resolvers
+
+- The external interface uses graphQL and these are handled with the resolvers.
+- The resolvers should contain no logic themselves. Rather this is delegated
+  to a `use-case` making the resolvers a thin wrapper around the controller logic.
+- Rationale is that it should be easy to implement another interface on the same
+  logic.
+
+#### Example
+
+```
+const { getSubmissionUseCase, } = require('use-cases')
+
+const resolvers = {
+  Query: {
+    async manuscript(_, { id }, { user }) {
+      return getSubmissionUseCase(id, user)
+    },
+    ...
+  }
+  ...
+```
+
+### Use-Cases
+
+- In general this layer will have an interface that maps one-to-one with the
+  graphQL layer.
+- For a given use-case it will require certain services and config - these
+  dependencies will be created by this layer and injected (when needed) into
+  the aggregates / models.
+
+#### Example
+
+```js
+const config = require('config')
+const Storage = require('MyStorageService')
+const Submission = require('aggregates/Submission')
+
+const User = require('@elifesciences/component-model').UserModel
+const File = require('@elifesciences/component-model').FileModel
+const Team = require('@elifesciences/component-model').TeamModel
+const Manuscript = require('@elifesciences/component-model').ManuscriptModel
+
+async getSubmissionUseCase(manuscriptId, user) => {
+  checkPermissions(id, user)
+  const models = { ... }
+  const services = { Storage, ... }
+  const submission = new Submision(config, models, services)
+  return submission.get(manuscriptId, user).toJSON()
+}
+
+module.exports = {
+  getSubmissionUseCase,
+}
+```
+
+### The Aggregate
+
+- [See DDD_Aggregates](https://www.martinfowler.com/bliki/DDD_Aggregate.html)
+- An aggregate
```

Comment:
should summarize the important principles from there:
- no persistent references to objects other than the Aggregate Root from outside the Aggregate (e.g. no fields referring to them and/or foreign keys, but passing them around for computation is ok)
- no transaction crosses Aggregates boundaries, as they are the unit of consistency; corollary: only eventual consistency cannot be guaranteed across Aggregate boundaries